src/internal/hooks/usePreferredColorScheme.ts

```diff
@@ -1,19 +1,22 @@
 import { useEffect, useState } from 'react';
 
 export function usePreferredColorScheme(): 'light' | 'dark' {
-  const mediaQuery = window.matchMedia('(prefers-color-scheme: dark)');
-  const [colorScheme, setColorScheme] = useState<'light' | 'dark'>(
-    mediaQuery.matches ? 'dark' : 'light',
-  );
+  const [colorScheme, setColorScheme] = useState<'light' | 'dark'>('light');
 
   useEffect(() => {
-    function updateColorScheme(event: MediaQueryListEvent) {
-      setColorScheme(event.matches ? 'dark' : 'light');
-    }
+    // Check if window is defined (client-side)
+    if (typeof window !== 'undefined' && window.matchMedia) {
```

Comment:
if you move this into the `useEffect`, do you still need to check the type? doesn't Next.js automatically use client-side rendering for all code inside React hooks?

Reply:
TIL